src/components/Header/NavComponents/chain-display.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useMemo } from "react";
+import {
+  Text,
+  useColorModeValue as mode,
+  Button,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverBody,
+  Stack,
+  PopoverArrow,
+  Center,
+} from "@chakra-ui/react";
+import { useChainAddresses } from "../../../utils/chainAddresses";
+import { changeChain } from "../../../utils/changeChain";
+import { internalAddressesPerNetworkId } from "../../../utils/contracts/contractAddresses/internalAddresses";
+import { UnsupportedChainIdError, useWeb3React } from "@web3-react/core";
+
+export const ChainSelector: React.FC<{}> = () => {
+  const chainAddresses = useChainAddresses();
+  const { error } = useWeb3React();
+
+  // Buttons to change to every available chain
+  const popoverData = useMemo(
+    () =>
+      Object.entries(internalAddressesPerNetworkId).map(([id, chain]) => {
+        return (
+          <Button
+            bg={mode({ base: "primary.500", md: "primary.500" }, "primary.500")}
+            colorScheme="teal"
+            size="xl"
+            h="40px"
+            onClick={() => changeChain(chain.chainName)}
+          >
+            {chain.chainName}
+          </Button>
+        );
+      }),
+    []
+  );
+
+  return (
+    <Popover>
+      <PopoverTrigger>
+        <Button
+          minWidth="9rem"
+          height={{ base: "4rem", md: "3rem" }}
+          fontSize={{ base: "4xl", md: "2xl" }}
+          mx="1.5rem"
+          px="1.5rem"
+          pt={{ base: "0px", md: "4px" }}
+          color="white"
+          bg={mode({ base: "secondary.800", md: "primary.500" }, "primary.500")}
+          rounded="lg"
+        >
+          <Text fontWeight="400">
+            {error && error instanceof UnsupportedChainIdError
+              ? `Invalid Chain`
+              : `Chain: ${chainAddresses?.chainName}`}
```

Comment:
This means that if `error` is defined *but* it isn't an `UnsupportedChainIdError`, you will still say the current chain as if there were no error. Is this what you wanted it to do? If so, it's fine.

Additionally, `<Button></Button>` should not contain `<Text></Text>`- instead, set the `fontWeight` parameter on `Button` itself, and drop the `Text` wrapper from around your `{error && ... }` section.

Similar advice applies anywhere else you've used `Text` members in `Button` components.